src/backend/optimizer/plan/subselect.c

```diff
@@ -785,7 +785,7 @@ build_subplan(PlannerInfo *root, Plan *plan, PlannerInfo *subroot,
 	 * there's no point since it won't get re-run without parameter changes
 	 * anyway.  The input of a hashed subplan doesn't need REWIND either.
 	 */
-	if (splan->parParam == NIL && !splan->is_initplan && !splan->useHashTable)
+	if (!splan->is_initplan)
```

Comment:
I think the comment above is better to update, so it is aligned with the `if` condition.